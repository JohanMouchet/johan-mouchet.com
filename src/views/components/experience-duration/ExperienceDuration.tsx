import { duration } from "@/utils/duration/duration";
import clsx, { ClassValue } from "clsx";
import styles from "./ExperienceDuration.module.scss";

export const ExperienceDuration = ({
  experiences,
  className,
  ...props
}: {
  experiences: {
    /** ISO string date */
    startDate: string;
    /** ISO string date */
    endDate: string;
  }[];
  className?: ClassValue;
} & React.HTMLProps<HTMLDivElement>) => {
  if (!experiences?.length) {
    return null;
  }

  return (
    <span
      className={clsx(styles["c-experience-duration"], className)}
      {...props}
    >
      {duration(
        experiences.map((experience: any) => {
          const startDate = new Date(experience.startDate);
          const endDate = experience.present
            ? new Date()
            : new Date(experience.endDate);

          return {
            start: startDate,
            end: endDate || new Date(),
          };
        })
      )}
    </span>
  );
};
