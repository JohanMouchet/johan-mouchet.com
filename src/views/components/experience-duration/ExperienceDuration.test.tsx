import { render } from "@testing-library/react";
import { ExperienceDuration } from "./ExperienceDuration";

it("Renders <ExperienceDuration /> without crashing", () => {
  expect(
    render(
      <ExperienceDuration
        experiences={[
          {
            startDate: new Date("2018-01-01"),
            endDate: new Date("2019-01-01"),
          },
        ]}
      />
    ).container.firstChild
  ).toMatchSnapshot();
});
