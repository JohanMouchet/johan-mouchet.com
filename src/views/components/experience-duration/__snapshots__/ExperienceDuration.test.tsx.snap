// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <ExperienceDuration /> without crashing 1`] = `
<span
  class="c-experience-duration"
>
  1 yr
</span>
`;
